src/transformers/visitors/toOrchestrationVisitor.ts

```diff
@@ -979,6 +983,8 @@ const visitor = {
   ElementaryTypeName: {
     enter(path: NodePath) {
       const { node, parent } = path;
+      if(!!path.getAncestorOfType('EventDefinition')) return;
+      if(!!path.getAncestorOfType('EmitStatement')) return;
```

Comment:
PR looks great, thank you!
One suggestion, could we add `EventDefinition` and `EmitStatement` with `enter` functions, then assign `state.skipSubNodes = true`? For here and the circuit visitor. It may be a bit cleaner, e.g.
```
  EventDefinition: {
    enter(path: NodePath, state: any) {
      state.skipSubNodes = true;
    }
  },
  ```

Reply:
Pushed new commit with subnodes skip. Thanks for suggestion!